sonarts-core/tests/rules/noTryPromiseRule/noTryPromiseRule.lint.ts

```diff
@@ -0,0 +1,91 @@
+export function someFunc() {
+
+}
+
+function returningPromise() {
+  return Promise.reject();
+}
+
+function singlePromise() {
+  try {
+//^^^ {{Consider using 'await' for the promise inside this 'try' or replace it with 'Promise.prototype.catch(...)' usage.}}
+    returningPromise();
+// [12:4-12:22] < {{Promise}}
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function conditionalPromise(cond: boolean) {
+  try {
+//^^^ {{Consider using 'await' for the promises inside this 'try' or replace it with 'Promise.prototype.catch(...)' usages.}}
+    if (cond) {
+      returningPromise();
+  // [23:6-23:24] < {{Promise}}
+    } else {
+      let x = 42;
+      returningPromise();
+  // [27:6-27:24] < {{Promise}}
+    }
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+async function okWithAwait() {
+  try {
+    await returningPromise();
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function okWithAnotherCall() {
+  try {
+    someFunc(); // can throw potentionally 
+    returningPromise();
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function okWithoutCatch() {
+  try {
+    returningPromise();
+  } finally {
+    console.log("finally");
+  }
+}
+
+async function severalTry() {
+  try {
+    await returningPromise();
+  } catch (e) {
+    console.log(e);
+  }
+
+  try {
+//^^^ {{Consider using 'await' for the promise inside this 'try' or replace it with 'Promise.prototype.catch(...)' usage.}}
+    returningPromise();
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function newPromise() {
+  try {
+//^^^ {{Consider using 'await' for the promise inside this 'try' or replace it with 'Promise.prototype.catch(...)' usage.}}
+    new Promise((res, rej) => {});
+// [78:4-78:33] < {{Promise}}
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function okWithNestedFunc() {
+  try {
+    let func = () => returningPromise();
+  } catch (e) {
+    console.log(e);
+  }
+}
```

Comment:
```typescript
function returningExceptionInsteadOfPromise() : Promise<number> {
  throw "foo";
}

// FP
function singlePromise() {
  try {
//^^^ {{Consider using 'await' for the promise inside this 'try' or replace it with 'Promise.prototype.catch(...)' usage.}}
    returningExceptionInsteadOfPromise();
  } catch (e) {
    console.log(e);
  }
}
```

We may have a FP when the function that constructs a promise throws an exception (and so the `try catch` would still be appropriate). What do you think?